crates/noirc_evaluator/src/ssa/conditional.rs

```diff
@@ -379,6 +379,7 @@ impl DecisionTree {
             // A second round is necessary when the synchronization optimizes function calls between the two branches.
             // In that case, the first cse updates the result instructions to the same call and then
             // the second cse can (and must) then simplify identical result instructions.
+            merged_ins.clear();
```

Comment:
Why are we clearing here? I assume there was a bug that was caught from doing cse twice with the same instruction lists, but we should add a comment to say why.

Reply:
Yes, obviously it was a bug as cse_block is writing the result to the block, so we should not start from the list again, but from the block instructions